web_external/ClipExplorer/index.jsx

```diff
@@ -0,0 +1,227 @@
+import _ from 'underscore';
+import React, { Component } from 'react';
+import { Modal, ButtonToolbar, Button, Row, Col, Glyphicon, Checkbox } from 'react-bootstrap';
+import { restRequest } from 'girder/rest';
+
+import style from './style.styl';
+
+class ClipExplorer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            paths: [{ name: 'root', folderId: 'root' }],
+            currentFolders: [],
+            rootFolders: null,
+            annotationFolders: new Set(),
+            selectedItem: null,
+            selectedFolderId: null,
+            showMode: 'all'
+        };
+        this.folderCache = new Map();
+    }
+
+    componentDidMount() {
+        this.fetchFolder();
+        this.getAllAnnotationFolders();
+    }
+
+    getAllFoldersAtRoot() {
+        return Promise.all([
+            restRequest({
+                url: '/collection',
+            }).then((collections) => {
+                return Promise.all(collections.map((collection) => {
```

Comment:
So... am I understanding this correctly, this will retrieve all root-level folders in *all* collections and present them as a single list? (What happens if some of these have the same name?)

Reply:
Yes, you understand right. I am aware of this. for now, it will work with two folders with the same name. I made this choice based on the data we distribute and the development time restraint.